src/builder/translators/translator_control_flow.cpp

```diff
@@ -9,122 +8,109 @@
 
 namespace builder {
 
-auto Translator::makeIfStatement(vhdlParser::If_statementContext *ctx) -> ast::IfStatement
+auto Translator::makeIfStatement(vhdlParser::If_statementContext &ctx) -> ast::IfStatement
 {
-    auto stmt = make<ast::IfStatement>(ctx);
-
-    // Main if branch
-    auto conditions = ctx->condition();
-    auto sequences = ctx->sequence_of_statements();
-
-    if (conditions.empty() || sequences.empty()) {
-        return stmt;
-    }
-
-    stmt.if_branch.condition = makeExpr(conditions[0]->expression());
-    stmt.if_branch.body = makeSequenceOfStatements(sequences[0]);
+    return build<ast::IfStatement>(ctx)
+      .with(&ctx,
+            [&](auto &node, auto &if_ctx) {
+                auto conditions = if_ctx.condition();
+                auto sequences = if_ctx.sequence_of_statements();
+
+                if (conditions.empty() || sequences.empty()) {
+                    return;
+                }
 
-    // elsif branches - number of elsif branches is conditions.size() - 1 (minus the initial if)
-    // If there's an else, the last sequence doesn't have a condition
-    for (const auto i : std::views::iota(std::size_t{ 1 }, conditions.size())) {
-        ast::IfStatement::Branch elsif_branch;
-        elsif_branch.condition = makeExpr(conditions[i]->expression());
-        elsif_branch.body = makeSequenceOfStatements(sequences[i]);
-        stmt.elsif_branches.emplace_back(std::move(elsif_branch));
-    }
+                // Main if branch
+                node.if_branch.condition = makeExpr(*conditions[0]->expression());
+                node.if_branch.body = makeSequenceOfStatements(*sequences[0]);
 
-    // else branch - if there are more sequences than conditions, the last one is else
-    if (sequences.size() > conditions.size()) {
-        ast::IfStatement::Branch else_branch;
-        // else has no condition - leave it empty
-        else_branch.body = makeSequenceOfStatements(sequences.back());
-        stmt.else_branch = std::move(else_branch);
-    }
+                // elsif branches
+                for (const auto i : std::views::iota(std::size_t{ 1 }, conditions.size())) {
+                    ast::IfStatement::Branch elsif_branch;
+                    elsif_branch.condition = makeExpr(*conditions[i]->expression());
+                    elsif_branch.body = makeSequenceOfStatements(*sequences[i]);
+                    node.elsif_branches.emplace_back(std::move(elsif_branch));
+                }
 
-    return stmt;
+                // else branch - if there are more sequences than conditions
+                if (sequences.size() > conditions.size()) {
```

Comment:
```suggestion
                if (std::cmp_greater(sequences.size(), conditions.size())) {
```

the bloated `std` offers an alternative here, not sure if this is actually better.